### ISSUE TITLE
Connecting with Tor - Captcha required

### ISSUE THREAD

**Issue by th3m** [EionRobb/skype4pidgin#343]
Since today i am unable to connect to skype using my pidgin plugin and Tor, i get this message in pidgin: "Captcha required. Try logging into web.skype.com and try again."

The same message appears in pidgin even after i successful logged to web.skype.com (with my Tor ip) using a browser.

**Comment by EionRobb** [comment]
Won't you be connecting via a new IP for each connection over Tor?    I'm not sure if the plugin is going to work over Tor like that.

**Comment by th3m** [comment]
Well the plugin works just fine over Tor, at the moment the Captcha thing is gone and i am able to connect as before. You may close this issue i guess. But it might come back.

When you connect to Tor you connect to 3 exit nodes and the last one gives you your IP. So no it doesn't change unless you change identity or restart Tor. The thing is that companies like Microsoft don't like Tor and occasionally try to limit the use of it by putting Captcha codes when they detect a potential try to login though it. Google also does this with its search engine, etc.

Your plugin works wonders to bring a level of anonymity and privacy even using skype, because you can connect through Tor in pidgin and use OTR. If there was not your plugin i would quit skype long time ago. Thank you

**Comment by Hexalyse** [comment]
You're wrong, @th3m 

More precisely, you do not connect to 3 exit nodes, you connect to 3 nodes, and one of them will serve as an exit node ! The other two can be relay nodes that do not enable exit policies.

And yes, as @EionRobb said, TOR periodically choose a new path of nodes in the network, so the exit node may change, hence your IP adress will change periodically for new connections.

You can find more informations regarding TOR on their official website, or on the Wikipedia page : https://en.wikipedia.org/wiki/Tor_%28anonymity_network%29

**Comment by th3m** [comment]
@Hexalyse Well you are correct, 3 relay nodes, 1 of them is the exit. But still it is a matter of terminology. Once you are connected to Skype, the Tor Circuit remains the same. Meaning that the exit node IP remains the same for that connection.

Still the problem of this issue remains the Captcha that may required so you can login to Skype-web. It happens occasionally now using Tor, it may appear for other reasons too. It may appear as a general safety measure. Then bye bye plugin?

**Comment by Hexalyse** [comment]
Yes, obviously someone must be using TOR to "crack/bruteforce/try" passwords of Skype accounts, so when you try to login, Skype asks for a captcha thinking you're the same person. Or maybe they have "greylisted" those exit nodes IP for some reason.

So... yes, I guess you can't use the plugin this way, unless the plugin provide a way to login with a captcha when Skype web service asks for one.

**Comment by Hernandy** [comment]
Same problem cant login, ask for Captcha. Connect using VPN

**Comment by black-roland** [comment]
The same problem. I use Spectrum Jabber transport on my own Jabber server. I tried to connect using Pidgin (from another IP address) and my Jabber transport asks for captcha now (if I connect from server IP).

> i login from web and change my password and suddenly working again.

Does not work for me. And my Skype account is blocked after that :) Be careful.

**Comment by EionRobb** [comment]
So I'm keen to get captcha login support going, but I'm gonna need some debug logs from people or a reliable way of getting the captcha's to trigger (other than "use tor") :)

**Comment by Hexalyse** [comment]
Maybe you can get the captcha by trying multiple wrong passwords ?
